applications/FluidDynamicsApplication/python_scripts/fluid_dynamics_analysis.py

```diff
@@ -49,11 +50,58 @@ def SetUpModel(self):
             self.solver.ImportModelPart()
             self.solver.AddDofs()
 
-        # Fill a Model instance using input
-        self.model = Kratos.Model()
         self.model.AddModelPart(self.main_model_part)
 
-    def SetUpAuxiliaryProcesses(self):
+        # this should let eventual derived stages modify the model after reading.
+        self.ModifyInitialProperties()
+        self.ModifyInitialGeometry()
+
+        self._SetUpListOfProcesses()
+        self._SetUpAnalysis()
+
+        for process in self.list_of_processes:
+            process.ExecuteBeforeSolutionLoop()
+
+    def RunSolutionLoop(self):
```

Comment:
I would say you should remove `RunSolutionLoop` and `Finalize` since they do exactly the same in the baseclass

Reply:
Ok but the `RunSolutionLoop` in the base class has a couple of minor syntax errors. I think I'll just replace it with mine.